public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -140,7 +145,7 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   );
 
   const simplifiedQueryStep =
-    isSwitchModeEnabled && isGrafanaAlertingType ? getValues('editorSettings.simplifiedQueryEditor') : false;
+    isSwitchModeEnabled && isGrafanaAlertingType ? watch('editorSettings.simplifiedQueryEditor') : false;
```

Comment:
this is necessary to keep the simpleCondition updated in this [useEffect](https://github.com/grafana/grafana/blob/d6c97bd6c964a9b22fad77f6b652b3fc9f944cff/public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx#L146-L150)

Reply:
`getValues` doesn't work as expected?
I wonder if we could simplify and stop rely on the `useEffect` hook if we used watch function with subcription callback https://react-hook-form.com/docs/useform/watch

Reply:
I've check and watch+callback would require too many changes.
We already watch the whole `editorSettings` object above. 

Would something like this work?
```suggestion
  const simplifiedQueryStep =
    isSwitchModeEnabled && isGrafanaAlertingType ? editorSettings?.simplifiedQueryEditor : false;
```